### ISSUE TITLE
M0 splitting and SliceReadoutTime length

### ISSUE THREAD

**Issue by jan-petr** [ExploreASL/ExploreASL#396]
Four related issues are fixed:

### Fix batch 1
- In BIDS2Legacy, the old version of `xASL_io_splitASL_M0` was called from `xASL_bids_parseM0`, needs to be renamed to `xASL_io_splitASL`
- In BIDS2Legacy, in `xASL_bids_parseM0`, when splitting M0 from ASL, we need to also split dummy-scans, or it won't be later possible as the scan indexes will change.
- In BIDS2Legacy, in `xASL_bids_parseM0` - after splitting M0 from ASL, the x.M0Type BIDS field needs to be removed so that it is not applied again.

### Fix batch 2
- In `xASL_quant_M0.m`, number of slices was not defined (nSlices) for the case where M0 acquisition at the end of TR was assumed.

### Fix batch 3
For SliceTime calculation, it always takes the slicetimereadout vector from x-struct. For both M0 and ASL. The issue is that in x-struct, the value is taken from ASL.json or studyPar.m. And we need to always load it from the corresponding M0.json and ASL4D.json, especially if the JSON values is different. That is an issue for a dataset with different number of slices in ASL and M0. To fix this, I need to do the following:
1. Give path or matrix to xASL_quant_SliceTimeVector.m, instead of just number of slices. Inside it calculates the number of slices form the matrix, or also loads the file, check its JSON sidecar and loads the SliceTime values.
2. In all three functions, xASL_quant_FEAST, Functions/xASL_quant_M0, Functions/xASL_quant_SinglePLD.m, we need to provide the path instead of the number of slices. 
3. xASL_wrp_ProcessM0.m also now provides the path to M0 rather than loaded M0 image.

### Fix batch 4
Minor typos in the function `xASL_quant_RevertBsupFxControl`.

### Release comment
No need to add anything to the release comment as this was only fixing issue in other new features. But we can add this issue number to SliceReadoutTime revamping. And to the M0-splitting.

**Comment by MauricePasternak** [comment]
I can confirm that the `xASL_quant_M0.m` issue is fixed for 2D acquisitions.

**Comment by MauricePasternak** [comment]
To confirm, is the MNI atlas no longer supported in the Population Module according to the `DataParTemplate.m` ?  Specifying `"MNI_structural"` in `x.S.Atlases` does not generate the .tsv files associated with that atlas.

Example of the DataPar section for that variable:

      "Atlases": [
         "TotalGM",
         "DeepWM",
         "MNI_structural"
      ]

**Comment by HenkMutsaerts** [comment]
> To confirm, is the MNI atlas no longer supported in the Population Module according to the `DataParTemplate.m` ? Specifying `"MNI_structural"` in `x.S.Atlases` does not generate the .tsv files associated with that atlas.
> 
> Example of the DataPar section for that variable:
> 
> ```
>   "Atlases": [
>      "TotalGM",
>      "DeepWM",
>      "MNI_structural"
>   ]
> ```

It's certainly still supported, Michael changed the behavior of this parameter perhaps? But it should indeed still be able to generate any atlas output

**Comment by HenkMutsaerts** [comment]
Actually, this is an issue to me, as MNI is a very general space, and the name of this atlas is MNI_structural. @MichaelStritt do you mind changing this? We also discussed to automate such that all atlas NIfTI file names are converted to field names for this parameter, but not implemented this yet /or halfway?

**Comment by HenkMutsaerts** [comment]
Very nice! Though we need to keep describing clearly what happens, keep function names self-explanatory, and make sure that everything is generalizable (e.g. that this function also works for ASL instead of M0, if needed).

**Comment by HenkMutsaerts** [comment]
See in-line comments. Should this issue be split (it essentially contains 2 issues)?

**Comment by jan-petr** [comment]
Answering all issue from above here:
1. I agree that this should have been 2 issues, but splitting will mean changing messages 15 commits back, changing IDs etc. So suboptimal, but not worth the effort - do you agree or still want to split?
2. Answered two inline comments [here](https://github.com/ExploreASL/ExploreASL/commit/8ad97b90e26470b37dbac7ea73c5cf4fbb659c13) - IMHO no need for code change.
3. Answered a comment [here](https://github.com/ExploreASL/ExploreASL/commit/90ccd4015b9d992fc5c2c96c57e57718b2c0f3c8). I agree with your comment on changing commit messages, no need to check.
4. Answered comments [here](https://github.com/ExploreASL/ExploreASL/commit/a0ae817e598bc4fc18b202d0598ccf54e39e2a1b) and made two commits for that. You might check both my answers and the commits, but I agree with all and made all the necessary changes.

If you agree with the final version, simply close this issue and move it to my bucket. Because I will then take all the release1.5.0 issues and updated the CHANGE.md in a single commit.

**Comment by jan-petr** [comment]
Excuse the two commits with the same name - I forgot to add a file in the first one. This only appears here - the commit history was reverted and is clean.

**Comment by jan-petr** [comment]
And one minor fix that crashed the splitting when M0 was not there, but Dummy yes.

**Comment by MichaelStritt** [comment]
> Actually, this is an issue to me, as MNI is a very general space, and the name of this atlas is MNI_structural. @MichaelStritt do you mind changing this? We also discussed to automate such that all atlas NIfTI file names are converted to field names for this parameter, but not implemented this yet /or halfway?

Let me remind you that you reviewed this :P 

Also this wasn't mentioned before. You wanted the filenames to be the field names, which is true for all the atlases in the Atlases directory.

I tested everything with the atlases I mentioned in the [description in the data par template](https://github.com/ExploreASL/ExploreASL/blob/develop/DataParTemplate.m).

You can also check out issue #56 and the [PR of issue #56](https://github.com/ExploreASL/ExploreASL/pull/245).

Long story short, we can easily change this. Check out the [definition in xASL_init_FileSystem](https://github.com/ExploreASL/ExploreASL/blob/develop/Functions/xASL_init_FileSystem.m). We only have to change `line 196` from `x.P.Atlas.MNI` to `x.P.Atlas.MNI_Structural`.

**Comment by HenkMutsaerts** [comment]
Yes, this sounds good. Apologies for not spotting this before. I'm completely happy with what you said, I would just keep the names of the atlases. Calling this atlas `MNI` is really confusing, as all atlases are in `MNI` space :)

**Comment by HenkMutsaerts** [comment]
So now this one is fixed right? Except for the incorrect naming (`SliceTiming` should still be `SliceReadoutTime` internally)?
